plugins/cilium-docker/main.go

```diff
@@ -50,6 +51,16 @@ connected to a Docker network of type "cilium".`,
   docker run --net my_network hello-world
 `,
 	Run: func(cmd *cobra.Command, args []string) {
+		if ciliumAPI == "" {
```

Comment:
`driver.NewDriver(ciliumAPI)` calls 

https://github.com/cilium/cilium/blob/1cf69804d818b88915a9c0b2b698b2c452706d2a/plugins/cilium-docker/driver/driver.go#L83

which calls
https://github.com/cilium/cilium/blob/e1850190e6e784aa1f72d554c1a3817ceaf776bd/pkg/client/client.go#L67-L76

Why is the code on `pkg/client/clieng.go` not being sufficient?

Reply:
There is no problem with pkg/client/client.go. The problem is scoppedLog is set with nil url if cilium-api isn't provided which ends up logging with no url.
e.g level=info msg="Waiting for cilium daemon to start up..." url=
https://github.com/cilium/cilium/blob/master/plugins/cilium-docker/driver/driver.go#L82

Reply:
@nirmoy oh, on that case I'll defer to @raybejjani

Reply:
I think most of the handling can be implemented by using `viper.BindEnv` (e.g. https://github.com/cilium/cilium/blob/master/daemon/main.go#L306) as part of setting up the options below. That makes it consistent (however cobra/viper defines consistent) with our other parameters that are both CLI and environment variables. The default can also be set directly there, so it will be printed in the help text.

@tgraf suggested something incompatible with what I said, so he will have to overrule what I'm saying (or not) to avoid causing confusion. If tgraf really wants the new function in client.go then a compromise would be to have that function be the "default" value of the parameter, implicitly doing all the logic it does.